### PR TITLE
Site Migration: Add a step to identify the target site platform

### PR DESCRIPTION
**Pull request by gabrielcaires**
Related to https://github.com/Automattic/wp-calypso/issues/87771

## Proposed Changes
* Add a step to verify the platform the user is using

**Step URL:**

`/setup/site-migration/site-migration-identify?siteSlug=[SITE_CREATED_ON_WP_COM]&flags=onboarding/new-migration-flow`

image

## Testing Instructions
The scenarios above require the creation of a site using the `setup/free` flow or any other flow. 

### Scenario 1:  WordPress site
1. Access the Step URL, replacing the  [SITE_CREATED_ON_WP_COM] placeholder with your site. [Example. `/setup/site-migration/site-migration-identify?siteSlug=site123.wpcomstaging.com&flags=onboarding/new-migration-flow`)
2.  Set the field `Where will you import from?` with the value `http://test-vanilla-wp.godaddy.test-wpcom-migrations.net/`. It is the site you want to migrate.
3. Click on the 'Continue' button
4.  Check the loading screen
5.  You should be redirected to the  `migrate-or-import step` (Screenshot 1) with the following params on the URL:
siteSlug=[SITE_CREATED_ON_WP_COM]
from=https://test-vanilla-wp.godaddy.test-wpcom-migrations.net/
siteID=XXXX
6. Continue the migration flow  (Import or MIgrate button)

### Scenario 2:  Site using any other platform  (Import content from another platform screen)
1. Access the Step URL, replacing the  [SITE_CREATED_ON_WP_COM] placeholder with your site. [Example. `/setup/site-migration/site-migration-identify?siteSlug=site123.wpcomstaging.com&flags=onboarding/new-migration-flow`)
2.  Set the field `Where will you import from?` with any site that is not using WordPress as the platform  (Example: ` adelinedimond.medium.com`)
4. Click on the 'Continue' button
5.  We should be redirected to the  `import content from another platform' (Screenshot 2)
6. Check if the URL contains the following attributes.
siteSlug=[SITE_CREATED_ON_WP_COM]
from=https://test-vanilla-wp.godaddy.test-wpcom-migrations.net/
siteID=XXXX
7. Continue the import flow (optional because no change was introduced).

---
### Screenshots

1. Scenario 1:   Migrate or Import step
image

2. Scenario 2: Import from another platform step
image

## Pre-merge Checklist

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-aUh-p2)?

### CODE REVIEW

**Comment by markbiek** [comment]
When I get to this point

CleanShot 2024-02-28 at 15 14 56

I'm taken to the "Import content from WordPress" screen if I click "WordPress".

CleanShot 2024-02-28 at 15 15 12

We had discussed before that clicking "WordPress" from the "Import content from another platform" should take the user to the Import vs. Migration step so they have another opportunity to migrate instead of importing.

CleanShot 2024-02-28 at 15 15 50

File: client/landing/stepper/declarative-flow/internals/steps.tsx [hunk 1]
```diff
@@ -211,6 +211,11 @@ export const STEPS = {
 		asyncComponent: () => import( './steps-repository/site-migration-instructions' ),
 	},
 
+	SITE_MIGRATION_CAPTURE: {
```
Comment (markbiek) [inline client/landing/stepper/declarative-flow/internals/steps.tsx]: Should this name be more consistent with the slug? Maybe `SITE_MIGRATION_IDENTIFY`?

File: client/landing/stepper/declarative-flow/internals/steps-repository/site-migration-identify/index.tsx [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+		</div>
+	);
+};
+
+const SiteMigrationIdentify: Step = function ( { navigation } ) {
+	return (
+		<>
+			<DocumentHead title="Site migration instructions" />
+			<StepContainer
+				stepName="site-migration-analyze"
```
Comment (markbiek) [inline client/landing/stepper/declarative-flow/internals/steps-repository/site-migration-identify/index.tsx]: I also feel like this naming should be consistent with the the step component name and slug. I don't have strong feelings if we use "identify" or "analyze" but I think we should use one or the other.

Comment (gabrielcaires) [reply client/landing/stepper/declarative-flow/internals/steps-repository/site-migration-identify/index.tsx]: Fixed here https://github.com/Automattic/wp-calypso/pull/87999/commits/b803f25b04c0fd5afbbfb5b69be58adbc46f6511

File: client/landing/stepper/declarative-flow/internals/steps.tsx (hunk 1 above)
Comment (gabrielcaires) [reply client/landing/stepper/declarative-flow/internals/steps.tsx]: Fixed here https://github.com/Automattic/wp-calypso/pull/87999/commits/b803f25b04c0fd5afbbfb5b69be58adbc46f6511

File: client/landing/stepper/declarative-flow/internals/steps-repository/site-migration-identify/index.tsx (hunk 2 above)
Comment (gabrielcaires) [reply client/landing/stepper/declarative-flow/internals/steps-repository/site-migration-identify/index.tsx]: Fixed here https://github.com/Automattic/wp-calypso/commit/b803f25b04c0fd5afbbfb5b69be58adbc46f6511

**Comment by gabrielcaires** [comment]
> We had discussed before that clicking "WordPress" from the "Import content from another platform" should take the user to the Import vs. Migration step so they have another opportunity to migrate instead of importing.
> 

I plan to do it as part of other PR since the code context is different.  @markbiek

**Review by markbiek — APPROVED** [review]
LGTM!

**Pull request merged**